src/extension.ts

```diff
@@ -257,18 +257,21 @@ export function activate(context: vscode.ExtensionContext): void {
         logPointsManager.toggleLogpoint(uri);
     });
 
-    context.subscriptions.push(vscode.commands.registerCommand('diagnostics.LogPoints.OpenScript', openScript));
+    initCommand(context, 'diagnostics.LogPoints.OpenScript', openScript);
 }
 
 // tslint:disable-next-line:no-empty
 export function deactivate(): void {
 }
 
-function initCommand<T>(extensionContext: vscode.ExtensionContext, commandId: string, callback: (context?: T) => void): void {
-    initAsyncCommand(extensionContext, commandId, async (context?: T) => callback(context));
+// tslint:disable-next-line:no-any
+function initCommand<T>(extensionContext: vscode.ExtensionContext, commandId: string, callback: (context?: T, ...rest: any[]) => void): void {
```

Comment:
It doesn't seem like the type parameter `T` is providing much value anymore and is just adding complexity

What if we just simplify this to be:
```
function initCommand(context: vscode.ExtensionContext, commandId: string, callback: (...args: any[]) => any)
```

Reply:
And then you can just call `await callback(...args);` down below

The cosmos repo has the simplified signature of `initCommand` and `initAsyncCommand`
https://github.com/Microsoft/vscode-cosmosdb/blob/master/src/extension.ts#L145

Reply:
I thought about that too. Not sure if there would be any special purpose of type T, I just wanted to minimize the impact to the codebase as much as possible.

Will change it in another PR.